dask_glm/tests/test_admm.py

```diff
@@ -46,11 +46,18 @@ def wrapped(beta, X, y, z, u, rho):
 @pytest.mark.parametrize('N', [1000, 10000])
 @pytest.mark.parametrize('nchunks', [5, 10])
 @pytest.mark.parametrize('p', [1, 5, 10])
-def test_admm_with_large_lamduh(N, p, nchunks):
+@pytest.mark.parametrize('is_cupy', [True, False])
+def test_admm_with_large_lamduh(N, p, nchunks, is_cupy):
     X = da.random.random((N, p), chunks=(N // nchunks, p))
     beta = np.random.random(p)
     y = make_y(X, beta=np.array(beta), chunks=(N // nchunks,))
 
+    if is_cupy:
+        cupy = pytest.importorskip('cupy')
+        X = X.map_blocks(lambda x: cupy.asarray(x),
+                         dtype=X.dtype, meta=cupy.asarray(X._meta))
+        y = y.map_blocks(lambda x: cupy.asarray(x),
+                         dtype=y.dtype, meta=cupy.asarray(y._meta))
```

Comment:
This looks very convoluted, the best way to generate `X` is:

```python
cupy = pytest.importorskip('cupy')
xp = cupy if is_cupy else np
rs = da.random.RandomState(RandomState=xp.random.RandomState)
X = rs.random((N, p), chunks=(N // nchunks, p))
```

It seems that `make_y` could also be changed to something such as:

```python
def make_y(X, beta=np.array([1.5, -3]), chunks=2, rs=da.random.RandomState(RandomState=np.random.RandomState)):
    n, p = X.shape
    z0 = X.dot(beta)
    y = rs.random(z0.shape, chunks=z0.chunks) < sigmoid(z0)
    return y
```

And then you'd call it similar to:

```python
y = make_y(X, beta=xp.array(beta), chunks=(N // nchunks,), rs=rs)
```

NOTE: I didn't actually test the code above, so it may need some fixing.

Reply:
You seem to be using this in 10 or so different tests, it's probably a good idea to write a small util to avoid copying it all back everywhere.

Reply:
Thank you for the review! I notice that `X` and `y` are generated in different ways in the current tests. The following utils are used:
- `X, y = make_intercept_data(N, p, seed=seed)`
- `X = da.random.random` and `y = make_y(...)`
- `X, y = make_classification(...)`
- `X, y = make_regression(...)`
- `X, y = make_poisson(...)`

I would like cupy inputs to be created using the same functions. Should I go into each of the utils above and add `is_cupy` branch? What I did currently is like just using one util function to convert the dask array to cupy after they are created.

Reply:
I see your point, maybe this is indeed too much work for very little benefit. With that said, I don't necessarily oppose to just leaving things as they are now, I'll defer to @mrocklin on whether he thinks it's acceptable. The only real downside I see with this is with people eventually referring to this code and thinking this is a good way to handle such cases, for tests this is ok but for real-world application this is not.

Reply:
Thank you. I wrapped up the repeated code in a util function to make it look nicer.